File: python/81_easy_number_challenge_236B.py
def divisor_count(x: int) -> int:
    cnt = 0
    d = 1
    while d * d <= x:
        if x % d == 0:
            cnt += 1
            if d * d != x:
                cnt += 1
        d += 1
    return cnt

n = int(input().strip())
ans = 0
for i in range(1, n + 1):
    for j in range(1, n + 1):
        for k in range(1, n + 1):
            ans += divisor_count(i * j * k)
print(ans % 1073741824)

File: c/81_easy_number_challenge_236B.c
#include <stdio.h>

static int divisor_count(int x) {
    int cnt = 0;
    for (int d = 1; d * d <= x; ++d) {
        if (x % d == 0) {
            ++cnt;
            if (d * d != x) ++cnt;
        }
    }
    return cnt;
}

int main(void) {
    int n;
    if (scanf("%d", &n) != 1) return 0;
    long long ans = 0;
    for (int i = 1; i <= n; ++i) {
        for (int j = 1; j <= n; ++j) {
            for (int k = 1; k <= n; ++k) {
                ans += divisor_count(i * j * k);
            }
        }
    }
    printf("%lld\n", ans % 1073741824LL);
    return 0;
}
